tests/kakuro/TestGameController.java

```diff
@@ -0,0 +1,130 @@
+// @author Brian Gamboc-Javiniar
+// @author Nalveer Moocheet
+// @author Vsevolod Ivanov
+// @author Chang Liu
+// @brief Test for the GameController class
+
+package kakuro;
+
+import static org.junit.Assert.*;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+
+import org.junit.Test;
+
+import kakuro.game.dao.GameDao;
+import kakuro.game.dao.GameDaoImpl;
+import kakuro.gameprogress.dao.GameProgressDaoImpl;
+import kakuro.utils.DatabaseConnection;
+
+public class TestGameController {
+
+    @Test
+    public void testGameController() {
+        //Arrange
+        int rows = 3;
+        int columns = 5;
+        //Act
+        GameController gameController = new GameController(columns,rows, false); 
+        //Set GUI to false
+        //Assert
+        assertEquals(columns, gameController.model.columns);
+        assertEquals(rows, gameController.model.rows);
+        
+    }
+    
+    @Test
+    public void testGetDatabaseConnection() {
+        DatabaseConnection db = new DatabaseConnection();
+        db.getConnection();
+    }
+    
+    @Test
+    public void testConnectDatabase() {
+        //Arrange
+        DatabaseConnection db = new DatabaseConnection();
+        db.connect();
+        //Act
+        Connection conn = db.getConnection(); 
+        //Assert
+        assertEquals(conn!=null, true);
+        db.disconnect();
+    }
+
+    @Test
+    public void testDisconnectDatabase() {
+        //Arrange
+        DatabaseConnection db = new DatabaseConnection();
+        db.connect();       
+        db.disconnect();
+        //Act
+        Connection conn = db.getConnection();
+        //Assert
+        try {
+          assertEquals(conn.isClosed(), true);
+        } catch (SQLException e) {
+          // TODO Auto-generated catch block
+          e.printStackTrace();
+        }         
+    }
+
+    @Test
+    public void testSolveBoard() {
+    Boolean GUI = false;
+    Boolean solved = false;        
+    // Arrange
+    int columns = 10;
+    int rows = 10;
+    // Act
+    GameModel model = new GameModel(columns, rows);
+    // Assert
+    assertEquals(model.columns, columns);
+    assertEquals(model.rows, rows);        
+    // Arrange
+    GameController controller1 = new GameController(10,10, GUI);
+    GameModel model1 = new GameModel(10,10); 
+    model1.initBoard();  //initialize board
+    model1.board[0][1] = new BoardCell(BoardCell.CellType.FILLED10, 5);
+    model1.board[1][0] = new BoardCell(BoardCell.CellType.FILLED01,-1,4);
+    model1.board[1][1] = new BoardCell(BoardCell.CellType.INPUT,3,-1);
+    model1.board[2][1] = new BoardCell(BoardCell.CellType.INPUT,2,-1);
+    model1.board[1][2] = new BoardCell(BoardCell.CellType.INPUT,1,-1);
+    controller1.model = model1;
+    // Act
```

Comment:
also shouldn't this be in three separates tests? the AAA part is one per each test